### ISSUE TITLE
raised from remote-0 protocol on 'XXX': expected '{' at 'BRBLOCK-1'

### ISSUE THREAD

**Issue by luza** [pgbackrest/pgbackrest#1044]
1. pgBackRest version: 2.26

2. PostgreSQL version: 9.6

3. Operating system/version:
database server - Debian 8
repository server - Ubuntu 16.04.6

4. pgBackRest installed from package

5. Config files
#### pgbackrest.conf on database server
```
[global]
log-level-file=detail
repo1-host=XXX
repo1-host-user=postgres

[pg1]
pg1-path=/var/lib/postgresql/9.6/main
```

#### pgbackrest.conf on repo server
```
[global]
repo1-retention-full=4
start-fast=y

repo1-type=s3
repo1-path=/pg1
repo1-s3-uri-style=path

repo1-s3-bucket=backup
repo1-s3-endpoint=YYY
repo1-s3-key=<redacted>
repo1-s3-key-secret=<redacted>
repo1-s3-region=ru-1a

[pg1]
pg1-host=ZZZ
pg1-path=/var/lib/postgresql/9.6/main
```

#### postgresql.conf
```
archive_command = 'pgbackrest --stanza=pg1 archive-push %p'
archive_mode = on
listen_addresses = '*'
max_wal_senders = 2
wal_level = replica
port = 5432
```

#### postgresql.log
```
ERROR: [099]: raised from remote-0 protocol on 'XXX': expected '{' at 'BRBLOCK-1'
2020-05-20 10:32:22.300 MSK [24760] LOG:  archive command failed with exit code 99
2020-05-20 10:32:22.300 MSK [24760] DETAIL:  The failed archive command was: pgbackrest --stanza=pg1 archive-push pg_xlog/000000010000030400000048
```

7. The issue:
The process of moving wal log files to repository server got stuck on a particular file with the message above. For now I managed to skip this file and archiving got back to normal.

What does this error message mean? How to prevent such cases in future?

**Comment by dwsteele** [comment]
This is generally caused by network issues. I'm not sure why it would happen multiple times for one file but there's no way to know without enabling more logging.

If it happens again increase the log level to debug before fixing the issue so we can see what's going on.

**Comment by luza** [comment]
I still have that file and can reproduce the error.

Attached the debug and trace-level output. Hope this will help.
[pgbackrest-debug.log](https://github.com/pgbackrest/pgbackrest/files/4659203/pgbackrest-debug.log)
[pgbackrest-trace.log](https://github.com/pgbackrest/pgbackrest/files/4659204/pgbackrest-trace.log)

**Comment by dwsteele** [comment]
That's helpful, thanks!

Here's the root error: `[500] Internal Server Error`. For some reason this is not causing an immediate abort but the protocol is left in a bad state, thus the weird error that comes next.

500 errors are retried, but we can see that all the retries failed due to the 60 second gap in the log waiting for the response:
```
2020-05-20 23:29:25.119 P00  TRACE:         common/io/handleRead::ioHandleRead: (this: {IoHandleRead}, buffer: {used: 0, size: 4194304, limit: <off>}, block: false)
2020-05-20 23:30:25.851 P00  TRACE:         common/io/handleRead::ioHandleRead: => 750
```
This is already a minor bug due to poor error reporting, but I'd really like to see if we can get more information about why S3 is complaining. Internal errors are generally random and for no special reason and I have never seen one be repeatable.

One theory -- AWS almost certainly uses some sort of hash to distribute load, so I wonder if this file keeps going to a broken server that has not yet been taken out of the fleet. If so, the problem will eventually self correct.

If we want to get more info about what's going on we'll need to look at the remote log. To get that set:
```
--log-level-file=trace --log-subprocess
```
Then on the repo host look for `/var/log/pgbackrest/pg1-archive-push-remote-001.log` and post that.

**Comment by luza** [comment]
I'm dealing not with AWS but with one of "third-party" S3 implementation. I guess that is kind of S3-to-Swift proxy because there OpenStack is used as the server platform.

Your assumption that the error is related to file name or its hash is correct, because after appending a couple of characters to file name the error vanishes. I am even able to reproduce error 500 by using `aws cli` tool for copying an empty file to `/pg1/archive/pg1/9.6-1/0000000100000304/000000010000030400000048-e78767ab531bc602d91f8f87d9d5da4cb7c41baa.gz` remote path.

So at this point I think I've better redirect the error report to the hoster. Thank you.

**Comment by dwsteele** [comment]
> I am even able to reproduce error 500 by using aws cli tool

Yeah, in that case better to talk to your provider.

I'm leaving this issue open because there's still an issue here with the error reporting in pgBackRest. I can see what's happening but I'm not sure about the best way to fix it.

**Comment by dwsteele** [comment]
I'm going to close this now because the reporting error should be fixed by #1119.
